drivers/gpio/gpio_smartbond.c

```diff
@@ -0,0 +1,286 @@
+
+	config->data_regs->data = value & mask;
+
+	return 0;
+}
+
+static int gpio_smartbond_port_set_bits_raw(const struct device *dev,
+					  gpio_port_pins_t pins)
+{
+	const struct gpio_smartbond_config *config = dev->config;
+
+	config->data_regs->set = pins;
+
+	return 0;
+}
+
+static int gpio_smartbond_port_clear_bits_raw(const struct device *dev,
+					    gpio_port_pins_t pins)
+{
+	const struct gpio_smartbond_config *config = dev->config;
+
+	config->data_regs->reset = pins;
+
+	return 0;
+}
+
+static int gpio_smartbond_port_toggle_bits(const struct device *dev,
+					 gpio_port_pins_t mask)
+{
+	const struct gpio_smartbond_config *config = dev->config;
+	volatile uint32_t *reg = &config->data_regs->data;
+
+	*reg = *reg ^ mask;
+
+	return 0;
+}
+
+static int gpio_smartbond_pin_interrupt_configure(const struct device *dev,
+						gpio_pin_t pin,
+						enum gpio_int_mode mode,
+						enum gpio_int_trig trig)
+{
+	const struct gpio_smartbond_config *config = dev->config;
+
+	/* Not supported by hardware */
+	if (mode == GPIO_INT_MODE_LEVEL) {
+		return -ENOTSUP;
+	}
+
+	if (mode == GPIO_INT_MODE_DISABLED) {
+		config->wkup_regs->sel &= ~BIT(pin);
+		config->wkup_regs->clear = BIT(pin);
+	} else {
+		if (trig == GPIO_INT_TRIG_BOTH) {
+			/* Not supported by hardware */
+			return -ENOTSUP;
+		}
+
+		if (trig == GPIO_INT_TRIG_HIGH) {
+			config->wkup_regs->pol &= ~BIT(pin);
+		} else {
+			config->wkup_regs->pol |= BIT(pin);
+		}
+
+		config->wkup_regs->sel |= BIT(pin);
+	}
+
+	return 0;
+}
+
+static int gpio_smartbond_manage_callback(const struct device *dev,
+				       struct gpio_callback *callback, bool set)
+{
+	struct gpio_smartbond_data *data = dev->data;
+
+	return gpio_manage_callback(&data->callbacks, callback, set);
+}
+
+static void gpio_smartbond_isr(const struct device *dev)
+{
+	const struct gpio_smartbond_config *config = dev->config;
+	struct gpio_smartbond_data *data = dev->data;
+	uint32_t stat;
+
+	WAKEUP->WKUP_RESET_IRQ_REG = WAKEUP_WKUP_RESET_IRQ_REG_WKUP_IRQ_RST_Msk;
+
+	stat = config->wkup_regs->status;
+	config->wkup_regs->clear = stat;
+
+	gpio_fire_callbacks(&data->callbacks, dev, stat);
+}
+
+/* GPIO driver registration */
+static const struct gpio_driver_api gpio_smartbond_drv_api_funcs = {
+	.pin_configure = gpio_smartbond_pin_configure,
+	.port_get_raw = gpio_smartbond_port_get_raw,
+	.port_set_masked_raw = gpio_smartbond_port_set_masked_raw,
+	.port_set_bits_raw = gpio_smartbond_port_set_bits_raw,
+	.port_clear_bits_raw = gpio_smartbond_port_clear_bits_raw,
+	.port_toggle_bits = gpio_smartbond_port_toggle_bits,
+	.pin_interrupt_configure = gpio_smartbond_pin_interrupt_configure,
+	.manage_callback = gpio_smartbond_manage_callback,
+};
+
+#define GPIO_SMARTBOND_DEVICE(id)							\
+	static const struct gpio_smartbond_config gpio_smartbond_p##id##_config = {	\
+		.common = {								\
+			.port_pin_mask =						\
+			GPIO_PORT_PIN_MASK_FROM_DT_INST(id),				\
```

Comment:
`GPIO_PORT_PIN_MASK_FROM_DT_INST()` is defined as:
```
#define GPIO_PORT_PIN_MASK_FROM_DT_INST(inst)                   \                                            
             GPIO_PORT_PIN_MASK_FROM_NGPIOS(DT_INST_PROP(inst, ngpios))
```
I couldn't find ngpios property in SoC's dts file.